### PR TITLE
Add pagination

### PR DESCRIPTION
**Pull request by Grasmus**

### CODE REVIEW

File: API/Startup.cs [hunk 1]
```diff
@@ -7,6 +7,8 @@
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
+using System.Text.Json;
+using System.Text.Json.Serialization;
```
Comment (marynakernychna) [inline API/Startup.cs]: Need to remove the extra usings

File: Core/Validation/PaginationFilterValidation.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Core.Validation
+{
+    public class PaginationFilterValidation : AbstractValidator<PaginationFilter>
+    {
+        public PaginationFilterValidation()
+        {
+            RuleFor(pf => pf.PageNumber)
+                .NotNull()
+                .GreaterThan(0).WithMessage("'{PropertyName}' should be greater than 0");
```
Comment (marynakernychna) [inline Core/Validation/PaginationFilterValidation.cs]: Maybe we should write "WithMessage" on a new line?

File: Core/Helpers/PaginatedList.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Core.Helpers
+{
+    public class PaginatedList<TEntity>
+    {
+        public int PageNumber { get; private set; }
+
+        public int TotalPages { get; private set; }
```
Comment (marynakernychna) [inline Core/Helpers/PaginatedList.cs:11]: Maybe we should also return the total count of items?

File: Core/Helpers/PaginatedList.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Core.Helpers
+{
+    public class PaginatedList<TEntity>
+    {
+        public int PageNumber { get; private set; }
+
```
Comment (marynakernychna) [inline Core/Helpers/PaginatedList.cs]: We do not write empty lines between properties

**Review by marynakernychna — APPROVED** [review]

**Review by vladtymo — APPROVED** [review]

File: Core/Services/CarService.cs [hunk 5]
```diff
@@ -56,12 +56,22 @@ public async Task<CarDTO> AddCarAsync(CreateCarDTO createCarDTO, string userId)
             return _mapper.Map<CarDTO>(car);
         }
 
-        public List<CarDTO> GetAllUserCars(string userId)
+        public async Task<PaginatedList<CarDTO>> GetAllUserCarsAsync(string userId, PaginationFilterDTO paginationFilter)
         {
-            var userCars = _carRepository
-                .GetIQuaryableBySpec(new CarSpecification.GetByUserId(userId));
+            var userCars = await _carRepository.ListAsync(new CarSpecification.GetByUserId(userId));
```
Comment (vladtymo) [inline Core/Services/CarService.cs:61]: Let's investigate how to optimize this query in order not to load all the items from db. It's might be a new ticket.

Comment (Grasmus) [reply Core/Services/CarService.cs:61]: It will be resolved in further pr

**Pull request merged**